todo_interface.rb

```diff
@@ -0,0 +1,165 @@
+require_relative 'todo_item_class.rb'
+require_relative 'todo_list_class.rb'
+require 'colorize'
+require 'date'
+
+
+def input_date
+	while(true)
+		puts "Now enter the due date in the form \"dd-mm-yyyy\": "
+		date_input = gets.chomp
+		d, m, y = date_input.split('-')
+		if Date.valid_date? y.to_i, m.to_i, d.to_i
+			begin
+			 	 return Date.parse(date_input)
+			 rescue Exception => e
+			 end
+		end
+		puts "Oops! Invalid date -- try again!".red
+	end
+end
+
+def input_priority
+	while(true)
```

Comment:
Same comment here re: the loop. Better to find a value or condition to use as the breakpoint.